packages/node-core/src/indexer/dictionary.service.ts

```diff
@@ -155,7 +156,10 @@ function buildDictQueryFragment(
 export class DictionaryService {
   private _client?: ApolloClient<NormalizedCacheObject>;
 
-  queriesMap?: BlockHeightMap<DictionaryQueryEntry[]>;
+  queriesMap?: BlockHeightMap<{
+    query: DictionaryQueryEntry[];
+    maxDsEndBlock: number; //maximum value of endBlock obtained from datasources for current project
```

Comment:
Block height map should already work out the end block https://github.com/subquery/subql/blob/main/packages/node-core/src/utils/blockHeightMap.ts#L4

We just need to change the entries in the map based on the end blocks